api/v1beta1/cluster_types.go

```diff
@@ -1073,6 +1088,7 @@ func (c *ClusterStatus) GetTypedPhase() ClusterPhase {
 // APIEndpoint represents a reachable Kubernetes API endpoint.
 type APIEndpoint struct {
 	// host is the hostname on which the API server is serving.
+	// +kubebuilder:validation:MaxLength=512
```

Comment:
Hostnames should be limited to 253 chars

Reply:
Are those actually only hostnames? 

> // APIEndpoint represents a reachable Kubernetes API endpoint.

Could these be longer?

Reply:
Potentially not then, this is why detailed descriptions are helpful

If this is not just the hostname and is actually a full endpoint with scheme and path, then 512 is more appropriate yes